lib/truffle/thread.rb

```diff
@@ -29,4 +29,21 @@ class ThreadError < StandardError
 class Thread
   Queue = ::Queue
   SizedQueue = ::SizedQueue
+
+  class Queue
+    def pop(non_block = false, timeout: nil)
+      if non_block
+        raise(ArgumentError, "can't set a timeout if non_block is enabled") unless Primitive.nil?(timeout)
+      else
```

Comment:
**subjective:** TBH I would prefer raising exception and the check for it to be a guard condition, I mean the following:
```ruby
if non_block && !Primitive.nil?(timeout)
  raise # ...
end
```

The only downside is that there is another `Primitive.nil?(timeout)` check below...

Reply:
While I prefer guards too this I'm not sure if worth - readability is kinda same. Was there a compiler/performance aspect to this? I know in some constructs a clear expression is better but here we're not doing that. If not - I would keep it this way.